src/ast/nodes/MetaProperty.ts

```diff
@@ -175,18 +175,17 @@ const relativeUrlMechanisms: Record<InternalModuleFormat, (relativePath: string)
 		return getResolveUrl(`require.toUrl('${relativePath}'), document.baseURI`);
 	},
 	cjs: relativePath =>
-		`(typeof document === 'undefined' ? ${getResolveUrl(
-			`'file:' + __dirname + '/${relativePath}'`,
-			`(require('u' + 'rl').URL)`
-		)} : ${getRelativeUrlFromDocument(relativePath)})`,
+		`(typeof document === 'undefined' ? ${`require('u' + 'rl').pathToFileURL(__dirname + '/${relativePath}').href`} : ${getRelativeUrlFromDocument(
```

Comment:
The generated code looks fine to me! One minor optimization to avoid duplication in this file: It would be nice to extract a helper function similar to `getResolveUrl` that encapsulates the logic and avoids repeating it in four places.

E.g.
```suggestion
		`(typeof document === 'undefined' ? ${getFileUrlFromDirname(relativePath)} : ${getRelativeUrlFromDocument(
```

and above

```ts
const getFileUrlFromDirname = (path: string) => `require('u' + 'rl').pathToFileURL(__dirname + '/${path}').href`;
```

Reply:
I added two helper functions:

https://github.com/rollup/rollup/blob/d10ffc3f15bd685f5caba869df72f99b2b8dd8a9/src/ast/nodes/MetaProperty.ts#L165-L168

These are now used to calculate the URL expressions from the full or relative filename.

Also fixed the failing test on Windows 🤞